core/src/main/scala/org/apache/spark/sql/execution/CommonUtils.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2017-2019 TIBCO Software Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package org.apache.spark.sql.execution
+
+import io.snappydata.Constant
+
+import org.apache.spark.sql.{SnappyParser, SnappySession}
+import org.apache.spark.sql.types.{CharType, DataType, MetadataBuilder, StringType, StructType, VarcharType}
+
+
+object CommonUtils {
+  def modifySchemaIfNeeded(options: Map[String, String], session: SnappySession,
+    schema: StructType): StructType = {
+    options.get("stringtypeas").map(stringTypeAs => {
```

Comment:
use 'string_type_as' instead of 'stringtypeas'. Also, a nit - this constant should be defined as a token somewhere else?